api/campaigns/class-maybe-show-campaign.php

```diff
@@ -76,30 +76,32 @@ function ( $post_data ) use ( $visit ) {
 		$referer_url                   = filter_input( INPUT_SERVER, 'HTTP_REFERER', FILTER_SANITIZE_STRING );
 		$page_referer_url              = isset( $_REQUEST['ref'] ) ? $_REQUEST['ref'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 		$all_segments                  = isset( $settings->all_segments ) ? $settings->all_segments : [];
-		$best_priority_segment         = $this->get_best_priority_segment( $all_segments, $client_id, $referer_url, $page_referer_url, $view_as_spec );
 		$overlay_to_maybe_display      = null;
 		$above_header_to_maybe_display = null;
 
-		// Filter campaigns to only those that match the best priority segment, or the default segment.
-		$matching_campaigns = array_filter(
-			$campaigns,
-			function( $campaign ) use ( $best_priority_segment ) {
-				return ( $best_priority_segment === $campaign->s || empty( $campaign->s ) );
-			}
-		);
+		/**
+		 * When previewing, spoof client ID so that user actions performed outside the preview window don't affect prompt display.
+		 * If a `cid` param is passed as part of the view_as spec, use that so it will persist across page loads within the preview session.
+		 * Otherwise, generate a unique ID for the current page view only.
+		 */
+		if ( ! empty( $view_as_spec ) ) {
+			$client_id = isset( $view_as_spec['cid'] ) ? $view_as_spec['cid'] : uniqid( 'view_as_client_' );
```

Comment:
It's not working as expected. When I preview, "once" frequency & dismissed prompts are displayed again on navigation. Client ID spoofing should probably implemented at a deeper level, either along with any occurrence of `CLIENT_ID()` AMP "function", or by replacing the cookie upon a preview request.
Either way, this can be tackled in a separate PR, too.

Reply:
> When I preview, "once" frequency & dismissed prompts are displayed again on navigation.

You're right, I see that even though the `cid` is persisting across navigations within the preview window, the client and campaign data aren't being saved. But this is true on `master` too, isn't it?

The main thing I'm trying to fix in this branch is that because we're no longer spoofing the actual segment (instead, we're just showing prompts if their assigned segments match the previewed segment), the actual client info is still being used to determine if prompts have been seen once or suppressed forever. What I see without a spoofed `cid`:

* As a logged in user, I have previously made test donations and subscribed to the newsletter. This is remembered by the preview session and can be confirmed by adding `$this->debug['client_data'] = $this->get_client_data( $client_id )['posts_read'];` to output the client data.
* As the same user, when previewing, this client data caused `suppress_forever` to be set to `true` on newsletter and donation prompts in the preview.
* Or, if I preview a segment with a "once" prompt, the prompt is recorded as having been seen once. If I then close the preview and preview again, the prompt isn't shown because I've already seen it.

For purposes of this PR, what do you think of keeping the spoofed `cid` so at least the expected prompts are seen with each preview, even if user actions and frequency aren't taken into account across navigation within the preview? Then, as you say, there might be a better way we can make previews match up to the front-end experience in another PR.

Reply:
Took a stab at spoofing the cookie in the plugin in https://github.com/Automattic/newspack-plugin/pull/891 & https://github.com/Automattic/newspack-popups/pull/458 - maybe this would solve the issue?

Reply:
Yeah, let's handle the cookie spoofing in those other two PRs. I've removed all spoofing behavior from this PR and https://github.com/Automattic/newspack-plugin/pull/874 for now.